operator/v1/types_kubeapiserver.go

```diff
@@ -24,6 +24,16 @@ type KubeAPIServerSpec struct {
 	// This provides a mechanism to kick a previously failed deployment and provide a reason why you think it will work
 	// this time instead of failing again on the same config.
 	ForceRedeploymentReason string `json:"forceRedeploymentReason"`
+
+	// certificateRotation contains information for overriding the default rotation times on particular certificates.
+	// You should only change settings here if you are very sure you know what you're doing. Changing defaults can
+	// cause cluster instability
+	CertificateRotation *KubeAPIServerCertificateRotationOverrides `json:"certificateRotation,omitempty"`
```

Comment:
I feel like this should be under an unsupported config section...

Reply:
> I feel like this should be under an unsupported config section...

That's a bold statement.  I would agree to do it, but it's bold.

Reply:
> > I feel like this should be under an unsupported config section...
> 
> That's a bold statement. I would agree to do it, but it's bold.

bold but I think Derek just agreed to it.

Reply:
i advocated for the need to make this configurable, not sure if it should be unsupported config yet.  if an organization had particular requirements around rotation that did not match our default, i would not want to make that an impedence to adoption of the platform.

Reply:
> i advocated for the need to make this configurable, not sure if it should be unsupported config yet. if an organization had particular requirements around rotation that did not match our default, i would not want to make that an impedence to adoption of the platform.

I strongly recommend against allowing critical, internal infrastructure like this be generally configurable.  I suspect you're likely to end up having a big mess if you do.  Comparing that choice against "it's cert rotation day", I may actually choose the latter instead.  That way at least I'll know I have a fairly fixed time bomb, not a series of differently configured ones at different places set at different times

Reply:
I agree with @enj and @deads2k that we want to have a config option for that, but making it supported is dangerous. We have learned how tricky it is to get rotation right, and I would bet that we have plenty of bug still in our code which show up if we allowed random rotation values.

Reply:
And worst: we cannot really test this properly, or even define required constraints for values to be sane and healthy for a cluster.

Reply:
Can we phrase this as "users are allowed to shorten the internal, but operator dictate what is the maximum allowed value for rotation". That means, QE or paranoid users can set rotation to 30 minutes (for testing), but if somebody set this to 2 days, operator will have the last word and have right to refuse this value.

Reply:
Can we then at least make it possible to allow a triggered cert rotation/renewal possible? this is not an unimaginable situation when facing a Disaster Recovery or a power (or other infrastructure) failure.